web/src/test/auto/integrated/cases/basics.spec.ts

```diff
@@ -0,0 +1,195 @@
+import { assert } from 'chai';
+
+import { DEVICE_DETECT_FAILURE, setupKMW, teardownKMW } from "../test_utils.js";
+import { Device } from "keyman/engine/device-detect";
+import { KeymanEngine } from "keyman/app/browser";
+
+const baseTimeout = 5000;
+
+const host = document.createElement('div');
+document.body.appendChild(host);
+
+describe('Basic KeymanWeb', function() {
+  this.timeout(baseTimeout);
+
+  before(function() {
+    // These tests require use of KMW's device-detection functionality.
+    assert.isFalse(DEVICE_DETECT_FAILURE, "Cannot run due to device detection failure.");
+  })
+
+  beforeEach(function() {
+    this.timeout(baseTimeout);
+
+    const singleton = document.createElement('input');
+    singleton.id = 'singleton';
+    host.appendChild(singleton);
+
+    return setupKMW(null, baseTimeout);
+  });
+
+  afterEach(function() {
+    host.innerHTML = '';
+    teardownKMW();
+  });
+
+  describe('Initialization', function() {
+    it('KMW should attach to the input element.', function() {
+      const keyman = window['keyman'] as KeymanEngine;
+      var singleton = document.getElementById('singleton');
+      assert.isTrue(keyman.isAttached(singleton), "KeymanWeb did not automatically attach to the element!");
+    });
+
+    it('KMW\'s initialization variable should indicate completion.', function() {
+      const keyman = window['keyman'] as KeymanEngine;
+      assert(keyman.initialized == 2, 'Keyman indicates incomplete initialization!');
```

Comment:
To better match the style of the other tests one could do:

```suggestion
      assert.equal(keyman.initialized, 2, 'Keyman indicates incomplete initialization!');
```

Reply:
Fair enough.  I'm pretty sure this was one of the first automated tests I ever wrote for KMW, hence how rough its definition is.  We just... never polished it up.  So yeah, let's do that now.